File: app/src/main/java/ru/henridellal/dialer/DialerApp.java
package ru.henridellal.dialer;

import android.app.Activity;
import android.app.Application;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import java.util.HashMap;
import java.util.Map;

public class DialerApp extends Application {
	
	public static final String LOG_TAG = "ru.henridellal.dialer";
	private static Map<String, Integer> themes = new HashMap<String, Integer>();
	static {
		themes.put("light", R.style.AppTheme_Light);
		themes.put("dark", R.style.AppTheme_Dark);
		themes.put("night", R.style.AppTheme_Night);
		themes.put("amoled", R.style.AppTheme_Amoled);
	}

	public static void setTheme(Activity activity) {
		SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(activity);
		String theme = preferences.getString("theme", "light");
		activity.setTheme(themes.get(theme));
	}

}

File: app/src/main/java/ru/henridellal/dialer/DialerApp.kt
package ru.henridellal.dialer

import android.app.Activity
import android.app.Application
import android.preference.PreferenceManager

object DialerApp : Application() {
    const val LOG_TAG = "ru.henridellal.dialer"
    @JvmStatic
	fun setTheme(activity: Activity) {
        val preferences = PreferenceManager.getDefaultSharedPreferences(activity)
        val theme = preferences.getString("theme", "light")
        if (theme == "light") {
            activity.setTheme(R.style.AppTheme_Dark)
        } else {
            activity.setTheme(R.style.AppTheme_Light)
        }
    }
}